04.ls/ls.rb

```diff
@@ -21,15 +21,27 @@ def format_output_strings(filenames)
   join_strings_by_row(rows, max_lengths_by_col)
 end
 
-def ls(args = ARGV)
+def parse_args_option(args)
   opt = OptionParser.new
   opt.on('-a')
+  opt.on('-r')
   options = {}
   args = opt.parse(args, into: options)
-  filepath = args[0] || '.'
+  [args, options]
+end
+
+def ls(args = ARGV)
+  parsed_args, options = parse_args_option(args)
+  filepath = parsed_args[0] || '.'
   filenames = Dir.entries(filepath).sort
   filenames = filenames.reject { |fname| fname.start_with?('.') } unless options[:a]
-  puts format_output_strings(filenames) unless filenames.empty?
+  filenames = filenames.reverse if options[:r]
+  "#{format_output_strings(filenames).join("\n")}\n" unless filenames.empty?
+end
+
+def ls_main(args = ARGV)
+  output_strings = ls(args)
+  print output_strings
 end
 
-ls if __FILE__ == $PROGRAM_NAME
+ls_main if __FILE__ == $PROGRAM_NAME
```

Comment:
ls_mainが大した仕事をしてないのでこれでいいかも？と思いました。

```suggestion
print ls(ARGV) if __FILE__ == $PROGRAM_NAME
```